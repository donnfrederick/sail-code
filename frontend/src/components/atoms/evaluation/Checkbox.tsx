import * as React from 'react'
import styled from 'styled-components'

import resolvePath from 'utils/resolvePath'

interface Props {
  isCenter?: boolean
  text: string
  isActive: boolean
  onSelect: () => void
  onUnselect: () => void
}
export default ({
  isCenter = false,
  text,
  isActive,
  onSelect,
  onUnselect
}: Props) => {
  const onClick = isActive ? onUnselect : onSelect
  return (
    <Container onClick={onClick} isCenter={isCenter}>
      <CheckboxContainer data-active={isActive}>
        <Checkbox
          data-active={isActive}
          src={resolvePath.image('common/evaluation/check@2x.png')}
        />
      </CheckboxContainer>
      <Description>{text}</Description>
    </Container>
  )
}

interface CenterProps {
  isCenter?: boolean
}

const Container = styled<CenterProps, any>('div')`
  height: 90px;
  width: 100%;
  margin-bottom: 10px;
  display: flex;
  ${({ isCenter }) => isCenter && 'justify-content: center'};
  align-items: center;
  background: none;
  color: rgb(65, 87, 101);
`

const CheckboxContainer = styled.div`
  position: relative;
  width: 44px;
  height: 44px;
  margin-right: 20px;
  box-sizing: border-box;
  border-radius: 8px;
  border: solid 3px rgb(158, 192, 204);
  background-color: white;
  &[data-active='true'] {
    border-color: transparent;
    background-color: rgb(19, 142, 254);
  }
`

const Checkbox = styled<{ isActive: boolean }, any>('img')`
  position: absolute;
  width: 23px;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  display: none;
  &[data-active='true'] {
    display: block;
  }
`

const Description = styled.span`
  font-size: 32px;
  font-weight: 500;
`
