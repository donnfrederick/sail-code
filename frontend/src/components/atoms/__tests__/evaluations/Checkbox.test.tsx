import Checkbox from 'components/atoms/evaluation/Checkbox'
import * as React from 'react'
import * as renderer from 'react-test-renderer'

test('render <Checkbox />', () => {
  const tree = renderer
    .create(
      <Checkbox
        isCenter={true}
        text="映像がたまに切れてた"
        isActive={true}
        onSelect={() => console.log('selected')}
        onUnselect={() => console.log('unselected')}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
