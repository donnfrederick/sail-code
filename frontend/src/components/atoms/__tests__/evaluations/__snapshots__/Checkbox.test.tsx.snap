// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <Checkbox /> 1`] = `
<div
  className="sc-bdVaJa euAaOh"
  onClick={[Function]}
>
  <div
    className="sc-bwzfXH kzjiuI"
    data-active={true}
  >
    <img
      className="sc-htpNat ldDEDY"
      data-active={true}
      src="/assets/img/common/evaluation/check@2x.png"
    />
  </div>
  <span
    className="sc-bxivhb jYwlOI"
  >
    映像がたまに切れてた
  </span>
</div>
`;
